tests/D2L.CodeStyle.Analyzers.Test/Specs/ImmutabilityAnalyzer.cs

```diff
@@ -54,6 +55,40 @@ public interface RegularInterface { }
 		[Immutable]
 		public interface SomeImmutableInterface { }
 
+		#region Constructor immutability
+		[Immutable]
+		public sealed class Good : RegularInterface { }
+
+		public class Bad : RegularInterface { }
+
+		[Immutable]
+		public sealed class SomeClassWithConstructor {
+			public readonly RegularInterface m_interface = new Good();
+
+			public SomeClassWithConstructor() {
+				m_interface = /* NonImmutableTypeHeldByImmutable(class, SpecTests.Types.Bad,  (or [ImmutableBaseClass])) */ new Bad() /**/;
+			}
+		}
+
+		[Immutable]
+		public sealed class SomeOtherClassWithConstructor {
+			public readonly RegularInterface m_interface = /* NonImmutableTypeHeldByImmutable(class, SpecTests.Types.Bad,  (or [ImmutableBaseClass])) */ new Bad() /**/;
+
+			public SomeClassWithConstructor() {
+				m_interface = /* NonImmutableTypeHeldByImmutable(class, SpecTests.Types.Bad,  (or [ImmutableBaseClass])) */ new Bad() /**/;
+			}
+		}
+
+		[Immutable]
+		public sealed class SomeOtherOtherClassWithConstructor {
+			public readonly RegularInterface m_interface = new Good();
+
+			public SomeClassWithConstructor() {
+				m_interface = new Good();
+			}
+		}
+		#endregion

```

Comment:
Other test cases that would be nice:

* one with a more complicated constructor, like one with writes inside an if statement (to catch the previous bug that `DescendantNodes()` fixed
* expression-bodies constructors (`public Mytype() => ...`)
* destructuring assignment

The last two might not work yet, and if so then they can be implemented in a follow-up (this PR has had enough back and forth :) ) I got this example from a Microsoft blog that hits the last two cases:

```cs
public record Person { 
    public string FirstName { get; init; } 
    public string LastName { get; init; }

    public Person(string firstName, string lastName) 
      => (FirstName, LastName) = (firstName, lastName);
}

Reply:
Maybe throw this in as a test case, it should pass now:

```cs
public record Person(string FirstName, string LastName);
```

It's equivalent to the previous, but it's constructor is implicit. We shouldn't analyze it's constructor, but that's fine because the writes inside of it are always safe.